Simplified/Catalog/NYPLCatalogUngroupedFeed.m

```diff
@@ -27,31 +27,55 @@ @interface NYPLCatalogUngroupedFeed ()
 // attempt to fetch more books will be made.
 static NSUInteger const preloadThreshold = 100;
 
+// The current number of attempts of books fetching until we receive supported books,
+// and the maximum attempts allowed.
+static NSInteger fetchSupportedBooksRetryCount = 0;
```

Comment:
is it possible to avoid using this static variable? The main problem is that these kind of variables have issues with multithreading. The other one (`fetchSupportedBooksRetryThreshold`) won't cause issues since it's constant, but it's not ideal either.

I think I know why you used them: the "fetch" method is static, so it's hard to keep a track of that state. Is it feasible to do one of these 2 things:
1. make `fetchSupportedBooksWithURL...` non static. If the method is a normal instance method, the "fetcher" instance can hold the value of the attempts being done so far. You will just need to manage ownership of the instance from wherever this is called from.
2. if `fetchSupportedBooksWithURL...` has to stay a class method, can you expand the signature to include the current iteration? that way you shouldn't need the `fetchSupportedBooksRetryCount` variable at all, since you'll be passing it along.

There's a third option, which is to just add a thread-safe protection to `fetchSupportedBooksRetryCount`, but I would not recommend it.

There's actually another problem maybe: what if this "fetch" method is called with a different URL while it's retrying? Can this happen, and if so shouldn't that use a separate retry counter? This is another argument from making this method a normal instance method.

Reply:
Right, thanks for catching the multithreading issue. The second option would be an easier fix, but I am going to explore the first option since it looks like a more appropriate implementation.